application/features/bootstrap/EmailContext.php

```diff
@@ -603,8 +620,9 @@ public function aBackupCodeMfaOptionWasXUsedDaysAgo($lastUsedDaysAgo)
     public function aBackupCodeIsUsedUpByThatUser()
     {
         $backupMfa = $this->getMfa(Mfa::TYPE_BACKUPCODE);
-        $backUpCode = MfaBackupcode::find()->where(['mfa_id' => $backupMfa->id])->one();
-        $backUpCode->delete();
+        $backUpCode = array_shift($this->tempBackupCodes);
+
+        MfaBackupcode::validateAndRemove($backupMfa->id, $backUpCode);
```

Comment:
One other thing probably worth doing is wrapping this in an `Assert::true(...)` to make sure the backup code was actually removed. I accidentally discovered the need for that when experimenting with these tests.

It may also be worth adding the `Assert::true(...)` where we call `validatedAndRemove(...)` up around line 594.

Reply:
I added this Assert but not the previous one, since there was a separate Assert  there to verify the total number of remaining backup codes.

Reply:
I added the Assert here but not in the earlier place, since it has a following Assert for the number of remaining codes.